blaze/server/tests/test_server.py

```diff
@@ -612,6 +612,61 @@ def test_bad_add_payload(temp_server, serial):
 
 
 @pytest.mark.parametrize('serial', all_formats)
+def test_add_expanded_payload(temp_server, serial):
+    # Ensure that the expanded payload format is accepted by the server
+    iris_path = example('iris.csv')
+    blob = serial.dumps({
+        'iris': {
+            'source': iris_path,
+            'extra': {'delimiter': ','}
+        }
+    })
+    response1 = temp_server.post('/add',
+                                 headers=mimetype(serial),
+                                 data=blob)
+    assert 'CREATED' in response1.status
+    assert response1.status_code == RC.CREATED
+
+
+@pytest.mark.parametrize('serial', all_formats)
+def test_add_expanded_payload_has_effect(temp_server, serial):
+    # Ensure that the expanded payload format actually passes the arguments
+    # through to the resource constructor
+    iris_path = example('iris-latin1.tsv')
+    csv_kwargs = {'delimiter': '\t', 'encoding': 'iso-8859-1'}
+    blob = serial.dumps({
+        'iris': {
+            'source': iris_path,
+            'extra': csv_kwargs
+        }
+    })
```

Comment:
Style nit (here and above) please reformat with the following style:

```python
blob = serial.dumps({'iris': {'source': iris_path,
                              'extra': csv_kwargs}})
```

We've settled on this indentation style for Blaze going forward, and are gradually updating files to conform.